doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+        targets as integers.
+
+        :func:`~utils.multiclass.type_of_target` will return 'multiclass' (or
+        'binary' in the degenerate case) for multiclass input.
+
+    multilabel
+        A multioutput target where each output is :term:`binary`.  This may be
+        represented as a 2d (dense) array or sparse matrix of integers, such
+        that each column is a separate binary target, where positive labels are
+        indicated with 1 and negative labels are usually -1 or 0.
+
+        Semantically, a multilabel target can be thought of as a set of labels
+        for each sample.  While not used internally,
+        :class:`preprocessing.MultiLabelBinarizer` is provided as a utility to
+        convert from a list of sets representation to a 2d array or sparse
+        matrix.
+
+        :func:`~utils.multiclass.type_of_target` will return
+        'multilabel-indicator' for multilabel input, whether sparse or dense.
+
+    multioutput continuous
+        TODO
+
+    multioutput multiclass
+        A classification problem .
+
+        TODO
+
+Methods
+=======
+
+.. glossary::
+
+    ``decision_function``
+
+        In a fitted :term:`classifier` or :term:`outlier detector`, predicts a
+        "soft" score for each sample in relation to each class, rather than the
+        "hard" categorical prediction produced by :term:`predict`.
+
+        Output conventions:
+
+        binary classification
+            A 1-dimensional array, where values strictly greater than zero
+            indicate the positive class (i.e. the last class in
+            :term:`classes_`).
+        multiclass classification
+            A 2-dimensional array, where the row-wise arg-maximum is the
+            predicted class.  Columns are ordered according to
+            :term:`classes_`.
+        multilabel classification
+            Scikit-learn is inconsistent in its representation of multilabel
+            decision functions.  Some represent it like multioutput multiclass,
+            i.e. a list of 2d arrays, each with two columns. Others represent
+            it with a single 2d array, whose columns correspond to the
+            individual binary classification decisions. The latter
+            representation is ambiguously identical to the multiclass
+            classification format, though its semantics differ: it should be
+            interpreted, like in the binary case, by thresholding at 0.
+
+            ..
+            see https://gist.github.com/jnothman/4807b1b0266613c20ba4d1f88d0f8cf5
+        multioutput classification
+            A list of 2d arrays, corresponding to each multiclass decision
+            function.
+        outlier detection
+            A 1-dimensional array, where a value greater than or equal to zero
+            (TODO: check equality case) indicates an inlier.
+
+    ``get_feature_names``
+        TODO
+
+    ``get_n_splits``
```

Comment:
that kinda belongs to a completely separate API, right? I'm not sure if it makes sense to put them into the same list.

Reply:
Sigh. I thought the same of "what do I do with `pos_label`"?